File: idea/testData/hierarchy/calls/callers/javaConstructor/main0.java
class KA {
    public <caret>KA() {

    }

    public final String name = "A";

    public String foo(String s) {
        return "A " + s;
    }
}

class KClient {
    {
        new KA();
    }

    public static final a = new KA();

    public final String getBar() {
        return new KA().name;
    }

    public final KA bar() {
        return new KA();
    }
}

File: kotlin-ide/idea/testData/hierarchy/calls/callees/kotlinClass/main0.kt
class KA {
    val name = "A"
    fun foo(s: String): String = "A: $s"
}

fun packageFun(s: String): String = s

val packageVal = ""

open class KClientBase {

}

class <caret>KClient(): KClientBase() {
    init {
        fun localFun(s: String): String = packageFun(s)


        KA().foo(KA().name)
        JA().foo(JA().name)
        localFun(packageVal)
    }

    val bar = run {
        val localVal = packageFun("")

        KA().foo(KA().name)
        JA().foo(JA().name)
        packageFun(localVal)
    }

    fun bar() {
        KA().foo(KA().name)
        JA().foo(JA().name)
    }
}